### PR TITLE
[TA2584] `MEMPOOL` command for ISTGTCONTROL to list istgt mempool and its stats

### PR DESCRIPTION
**Pull request by mynktl**
Changes : 
- Added `MEMPOOL` command to list memory pool(allocated for replication) and its stats. It will also list `spec` in-flight IOs count and `replica` in-flight IOs count.
Command: `istgtcontrol MEMPOOL`
The output will be in JSON format as listed below: 

```jsonc
{
  "mempool list": [
    {
      "MEMPOOL": "rcommon_mempool",
      "Size": 100000,
      "Free": 100000,
      "volume usage": [
        {
          "volume": "vol1",
          "in-flight read": 0,
          "in-flight write": 0,
          "in-flight sync": 0
        }
      ]
    },
    {
      "MEMPOOL": "rcommon_mempool",
      "Size": 300000,
      "Free": 300000,
      "replica usage": [
        {
          "replica": 6061,
          "in-flight read": 0,
          "in-flight write": 0,
          "in-flight sync": 0
        },
        {
          "replica": 6062,
          "in-flight read": 0,
          "in-flight write": 0,
          "in-flight sync": 0
        },
        {
          "replica": 6063,
          "in-flight read": 0,
          "in-flight write": 0,
          "in-flight sync": 0
        }
      ]
    }
  ]
}
```

### CODE REVIEW

**Review by utkarshmani1997 — APPROVED** [review]
LGTM

File: src/replication.c [hunk 1]
```diff
@@ -2959,6 +2968,96 @@ destroy_replication_mempool(void)
...
+
+	j_obj = json_object_new_object();
+	json_object_object_add(j_obj, "mempool list", j_resp);
+
+	json_string = json_object_to_json_string_ext(j_obj,
+	    JSON_C_TO_STRING_PLAIN);
+	resp_len = strlen(json_string);
+	*resp = malloc(resp_len);
+	memset(*resp, 0, resp_len);
+	strncpy(*resp, json_string, resp_len);
```
Comment (vishnuitta) [inline src/replication.c:3154]: what about free of this?

File: src/replication.c [hunk 2]
```diff
@@ -2959,6 +2968,96 @@ destroy_replication_mempool(void)
...
+	MTX_UNLOCK(&specq_mtx);
+	json_object_object_add(j_obj, "replica usage", j_array);
+	json_object_array_add(j_resp, j_obj);
+
+	/* rcmd_mempool end */
+
+	j_obj = json_object_new_object();
+	json_object_object_add(j_obj, "mempool list", j_resp);
+
+	json_string = json_object_to_json_string_ext(j_obj,
```
Comment (vishnuitta) [inline src/replication.c:3149]: what about freeing of this string?

File: src/replication.c (hunk 1 above)
Comment (vishnuitta) [reply src/replication.c:3154]: fine.. this is taken care in caller

**Review by vishnuitta — APPROVED** [review]
changes are good

**Comment by vishnuitta** [comment]
@mynktl please resolve conflicts

**Comment by mynktl** [comment]
@vishnuitta I have resolved conflicts and also removed few unused variables from `istgt_lu_ctl.c` from previous merged PR.

**Pull request merged**